### PR TITLE
raft: clean up

### PR DESCRIPTION
**Pull request by BusyJay**
@ngaut @siddontang @disksing @zhangjinpeng1987 PTAL

### CODE REVIEW

File: src/raft/raft_log.rs [hunk 1]
```diff
@@ -304,11 +303,7 @@ impl<T: Storage> RaftLog<T> {
     }
 
     pub fn snapshot(&self) -> Result<Snapshot> {
-        if self.unstable.snapshot.is_some() {
-            Ok(self.unstable.get_snapshot())
```
Comment (siddontang) [inline src/raft/raft_log.rs]: can we remove get_snapshot in log unstable now?

Comment (BusyJay) [inline src/raft/raft_log.rs]: It was removed.

**Comment by BusyJay** [comment]
ping

**Comment by ngaut** [comment]
LGTM

**Comment by BusyJay** [comment]
@siddontang @disksing @zhangjinpeng1987 PTAL

**Comment by siddontang** [comment]
LGTM

**Pull request merged**